src/kaleidoscope_internal/event_dispatch.h

```diff
@@ -167,7 +167,7 @@
    result = EventHandler__::call(PLUGIN, hook_args...);              __NL__ \
                                                                      __NL__ \
    if (EventHandler__::shouldAbortOnConsumedEvent() &&               __NL__ \
```

Comment:
Before I read this change, I was wondering if we should instead be creating a method like

```EventHandler__::shouldAbortOnResult(kaleidscope::EventHandlerResult)``` and move the logic in there.



Now that I'm looking, while I understand the intent for the change from == EVENT_CONSUMED to != OK,  it makes the conditional of 

```
if (EventHandler__::shouldAbortOnConsumedEvent() && result != kaleidoscope::EventHandlerResult::OK) {             ```
 read kind of funny, which makes me think that moving the check into a method makes sense.

Reply:
I'm in complete agreement. This was a case of me trying to make the change as small as possible to make it as easy to verify as correct as possible. I don't have a sufficiently complete understanding of how the plugin hook system works to be confident in changing it, but the "consumed event" in the name definitely starts to become a lie with the change I made here.

Maybe it would be best to do that as a separate PR from this one?

Reply:
I added a small commit changing the name `shouldAbortOnConsumedEvent()` to `shouldExitIfResultNotOk()`. I know that's not really what you had in mind, but it at least matches the new conditional better. I think it might be best to avoid use of the same term ("abort", unless you want to change `EventHandlerResult::ABORT` to something else—`CANCEL`, perhaps? (I still think "abort" is more appropriate)) in both places, and I think "exit" or "return" would be more descriptive of what the hook function will do if it gets a non-`OK` result from a handler.

Reply:
I tried to make the change you suggested, but I get an inscrutable error in preprocessor macro expansion:
```
error: unterminated argument list invoking macro "EVAL4"
```
I've scrutinized the three lines of code I changed, and have no idea how to resolve that, so I'm giving up for now.